src/PostgREST/QueryBuilder.hs

```diff
@@ -0,0 +1,182 @@
+module PostgREST.QueryBuilder
+where
+
+
+import           Control.Error
+import           Data.List         (find)
+import           Data.Monoid
+import           Data.Text         hiding (filter, find, foldr, head, last, map,
+                                    null)
+import           Control.Applicative
+import           Data.Tree
+import           PostgREST.PgQuery (PStmt, QualifiedIdentifier (..), fromQi,
+                                    orderT, pgFmtIdent, pgFmtLit, pgFmtOperator,
+                                    pgFmtValue, whiteList)
+import           PostgREST.Types
+--import qualified Hasql as H
+--import qualified Hasql.Postgres as P
+--import Control.Applicative ((<|>))
+import qualified Data.Vector       as V (empty)
+import qualified Hasql.Backend     as B
+
+findColumn :: [Column] -> Text -> Text -> Text -> Either Text Column
+findColumn allColumns s t c = note ("no such column: "<>t<>"."<>c) $
+  find (\ col -> colSchema col == s && colTable col == t && colName col == c ) allColumns
+
+findTable :: [Table] -> Text -> Text -> Either Text Table
+findTable allTables s t = note ("no such table: "<>t) $
```

Comment:
I don't see `findColumn` or `findTable` used in other parts of the code. Are they here for future convenience?